packages/amplify-graphql-schema-generator/src/ts-schema-generator/ts-schema-generator.ts

```diff
@@ -54,4 +56,17 @@ export class TypescriptDataSchemaGenerator {
     models.forEach((m) => schema.addModel(m));
     return schema;
   };
+
+  private static createIdentifier = (config: TypescriptDataSchemaGeneratorConfig): string => {
+    const { host, database, engine, port } = config;
+    const identifierString = host.concat(engine, database, port.toString());
+    const identifierHash = createHash('md5').update(identifierString).digest('base64');
```

Comment:
Do we need to be using MD5? I know we're not doing anything particularly sensitive, but it'll inevitably cause false positives in security scanners that we'll have to address. SHA2 would be a better choice if the length constraints support it (which we should validate with tests). If not, add a doc comment here to explain how we're using MD5 and why it's not a security-sensitive use case

Reply:
Added comments explaining why we use 'md5' here. Considering our intention here is to generate an unique id and not to check the integrity of the data, it is okay to use 'md5'. Also, we only use the hostname, engine, port number and database to generate the ID, this doesn't post a security threat.